nodes/modifier_make/contour2D.py

```diff
@@ -42,58 +42,59 @@
     ("Poligonal", "Poligonal", "Intersecction dependent from num. of vertices (Faster)", 1)]
 
 
-def check_dist_to_verts(v, or_verts, or_radius, net, poli_ang_list, modulo, mask_t):
+def check_dist_to_verts(v, or_verts, or_radius, net, poli_ang_list, poly_ang_cos_list, mask_t):
     d = 0
     min_dist = 2.0e-5
-    for j in range(modulo):
-        rad = or_radius[j]
+    for rad, vo, polig_ang, polig_ang_cos, v_net in zip(or_radius, or_verts, poli_ang_list, poly_ang_cos_list, net):
         if rad < min_dist:
             continue
-        vo = or_verts[j]
+
         vfx = v[0]-vo[0]
         vfy = v[1]-vo[1]
-        # dist_to_point = Vector((vfx, vfy)).length
+
         dist_to_point = length_v2((vfx, vfy))
         if dist_to_point > rad:
             continue
 
-        polig_ang = poli_ang_list[j]
-        offset = net[j][1][0]
-        if len(net[j][0]) == 3:
-            offset += net[j][0][0] - net[j][0][1]
+        offset = v_net[1][0]
+        if len(v_net[0]) == 3:
+            offset += v_net[0][0] - v_net[0][1]
         v_vo_ang = (normal_angle(atan2(vfy, vfx) - offset) % (2 * polig_ang)) - polig_ang
```

Comment:
Maybe it is possible to get rid of trigonometry functions here? Like, cos(atan2(vfy,vfx)) = vfx
If not, then vectorization with numpy can probably help.

Reply:
Thanks for the hints, I am doing a deeper rewrite to improve the performance of this node because it is still very slow and I will consider your ideas :)